bizflycloud/resource_bizflycloud_kubernetes.go

```diff
@@ -0,0 +1,234 @@
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"enable_autoscaling": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"tags": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Required: true,
+			},
+			"auto_upgrade": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"enable_cloud": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"create_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"worker_pools": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Resource{Schema: workerPoolSchema()},
+			},
+			"worker_pools_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func workerPoolSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"version": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"flavor": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"profile_type": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"volume_type": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"volume_size": {
+			Type:     schema.TypeInt,
+			Required: true,
+		},
+		"available_zone": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"desire_size": {
+			Type:     schema.TypeInt,
+			Required: true,
+		},
+		"enable_autoscaling": {
+			Type:     schema.TypeBool,
+			Required: true,
+		},
+		"min_size": {
+			Type:     schema.TypeInt,
+			Required: true,
+		},
+		"max_size": {
+			Type:     schema.TypeInt,
+			Required: true,
+		},
+		"tags": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem:     &schema.Schema{Type: schema.TypeString},
+		},
+	}
+}
+func resourceBizFlyClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*CombinedConfig).gobizflyClient()
+
+	// Build up creation options
+	ccrq := &gobizfly.ClusterCreateRequest{
+		Name:        d.Get("name").(string),
+		Version:     d.Get("version").(string),
+		AutoUpgrade: d.Get("auto_upgrade").(bool),
+		EnableCloud: d.Get("enable_cloud").(bool),
+		Tags:        d.Get("tags").([]string),
+		WorkerPools: d.Get("worker_pools").([]gobizfly.WorkerPool),
+	}
+	log.Printf("[DEBUG] Create Cluster configuration: %#v\n", ccrq)
+	cluster, err := client.KubernetesEngine.Create(context.Background(), ccrq)
+	if err != nil {
+		return fmt.Errorf("Error creating cluster: %v", err)
+	}
+	d.SetId(cluster.UID)
+	err = resourceBizFlyCloudClusterRead(d, meta)
+	if err != nil {
```

Comment:
đoạn này ở dưới resourceBizFlyCloudClusterRead có rồi nên chỉ cần return function là đủ.

Reply:
nếu không nhầm thì lỗi return và in trên màn hình sẽ là: Error creating cluster: Error creating cluster: xxxx